recipes/mpfr/all/test_v1_package/CMakeLists.txt

```
@@ -0,0 +1,8 @@
+cmake_minimum_required(VERSION 3.3.0)
+project(test_package C)
+
+include("${CMAKE_BINARY_DIR}/conanbuildinfo.cmake")
+conan_basic_setup()
+
+add_executable(${PROJECT_NAME} test_package.c)
+target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
```

```suggestion
conan_basic_setup(TARGETS)

add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/../test_package/
                 ${CMAKE_CURRENT_BINARY_DIR}/test_package/)
```
please, use cmake targets to validate those properties from your recipe.

---

Hi @uilianries,
I applied this suggestion but had to do it manually, because I had changed the path to test_package.c before I realized that this change encompasses that.

---

Hi @uilianries,
I tried the suggested change and numerous variations but was unable to get any of them to work. Admittedly I am not very familiar with Conan 1.x as my work has been focused on Conan 2.0, but I don't see how 
add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/../test_package/) could ever work given that ../test_package has its own CMakeLists.txt file and this command appears to cause execution to pass to that one. I was able to finally get things working by going back to something closer to what I had originally, with the exception that I am using the TARGETS argument to conan_basic_setup, which causes CONAN_PKG::mpfr to be defined so I use that in the target_link_libraries function. Also, I reference the version of test_package.c in the test_package directory by using ../test_package as a relative path. If I have missed something here or there is some other way to do what you were suggesting please let me know. Thanks,